### DISCUSSION TITLE
"Session did not start before time expired." for a deployed application

### DISCUSSION THREAD

**Discussion by Sichao25** [Kitware/trame#520] [Q&A] [answered]
Hi!

I deployed my trame application with the caprover. The deployment itself finished successfully. However, the web application will keep loading until it times out. I have already set timeout=0 to avoid auto shutdown. The logs are:

```
 Error handling request
| Traceback (most recent call last):
|   File "/deploy/server/venv/lib/python3.9/site-packages/aiohttp/web_protocol.py", line 452, in _handle_request
|     resp = await request_handler(request)
|   File "/deploy/server/venv/lib/python3.9/site-packages/aiohttp/web_app.py", line 543, in _handle
|     resp = await handler(request)
|   File "/deploy/server/venv/lib/python3.9/site-packages/wslink/backends/aiohttp/launcher.py", line 97, in handle_post
|     return await self._waitForReady(session, request)
|   File "/deploy/server/venv/lib/python3.9/site-packages/wslink/backends/aiohttp/launcher.py", line 109, in _waitForReady
|     if self.process_manager.isReady(session, count):
|   File "/deploy/server/venv/lib/python3.9/site-packages/wslink/launcher.py", line 480, in isReady
|     if not self.isRunning(id) and count < 60:
|   File "/deploy/server/venv/lib/python3.9/site-packages/wslink/launcher.py", line 468, in isRunning
|     return self.processes[id].poll() is None
| KeyError: 'd905b752-0f26-11ef-938c-02420a000121'
```
Any suggestions on what this means?

Thanks in advance.

**Comment by jourdain** [comment]
Where did you set that timeout 0? Also that error is either because the launcher did not detect that the app was ready, or that the app crashed. Can you look at the session logs in `/deploy/server/logs/launcher/*.txt`.

**Reply by Sichao25** [reply]
Thanks for the response. At first, I thought it could be related to auto-shutdown, which is why I set the timeout to 0. Now I checked the session logs and found the error `ModuleNotFoundError: No module named 'tkinter'.`

**Reply by jourdain** [reply, marked as answer]
You can not use Tk in a pure headless server. Everything needs to be web to work.

**Reply by Sichao25** [reply]
Thanks for the clarification!
